sd/lb/retry.go

```diff
@@ -1,57 +1,83 @@
 package lb
 
 import (
-	"fmt"
-	"strings"
-	"time"
+    "fmt"
+    "strings"
+    "time"
 
-	"golang.org/x/net/context"
+    "golang.org/x/net/context"
 
-	"github.com/go-kit/kit/endpoint"
+    "github.com/go-kit/kit/endpoint"
 )
 
+// Callback function that indicates the current attempt count and the error encountered.
+//  Should return whether the Retry function should continue trying, and a custom 
+//  error message if desired. The error message may be nil, but a true/false 
+//  is always expected. In all cases if the error message is supplied, the 
+//  current error will be replaced.
+type Callback func(int, string) (bool, *string)
```

Comment:
Hmm, a pointer to a string is not a great way to encode maybe-error. It would make a lot more sense if Callback had the signature

```go
type Callback func(n int, err error) (continue bool, err error)
```